### DISCUSSION TITLE
Help with a Multiclass Example

### DISCUSSION THREAD

**Discussion by muriloasouza** [blue-yonder/tsfresh#1015] [Q&A] [answered]
I need to solve a Multiclass Problem, and i have been following the Jupyter Notebook of the [Multiclass Selection Example](https://github.com/blue-yonder/tsfresh/blob/main/notebooks/04%20Multiclass%20Selection%20Example.ipynb).

My dataset is a bit different, each `id` has `48` timesteps (half-hourly measurments of a single day), but i have `N` days of measurements for each `id`.

The values of my `time` column are `5-digit` numbers, where the first `3` digits represents the day, and the last `2` digits represents the `timesteps`. Something like this:

**Example1:** The first row is the `ID = 0` and have the code  `time=19501`. So the `195` is a number that represents the first day of recorded data and the `01` part represents the time from 00:00:00 to 00:29:59;

**Example2**: In the second row i have the code `19502` which is the first day from 00:30:00 to 00:59:59 to `ID=0`;

So, the last three digits alway range from `01~48` (48 half-hourly measurements). Here is the dataset:

```
      ID   Time  Consumption (kWh)
0   1000  19501              0.117
1   1000  19502              0.050
2   1000  19503              0.025
3   1000  19504              0.025
4   1000  19505              0.049
5   1000  19506              0.117
6   1000  19507              0.042
7   1000  19508              0.024
8   1000  19509              0.024
9   1000  19510              0.048
10  1000  19511              0.114
11  1000  19512              0.039
12  1000  19513              0.024
13  1000  19514              0.024
14  1000  19515              0.049
15  1000  19516              0.113
16  1000  19517              0.036
17  1000  19518              0.102
18  1000  19519              0.114
19  1000  19520              0.092
20  1000  19521              0.029
21  1000  19522              0.030
22  1000  19523              0.030
23  1000  19524              0.090
24  1000  19525              0.111
25  1000  19526              0.030
26  1000  19527              0.029
27  1000  19528              0.029
28  1000  19529              0.079
29  1000  19530              0.119
30  1000  19531              0.030
31  1000  19532              0.031
32  1000  19533              0.031
33  1000  19534              0.079
34  1000  19535              0.098
35  1000  19536              0.029
36  1000  19537              0.028
37  1000  19538              0.024
38  1000  19539              1.597
39  1000  19540              3.453
40  1000  19541              1.288
41  1000  19542              0.102
42  1000  19543              0.789
43  1000  19544              0.195
44  1000  19545              0.129
45  1000  19546              1.133
46  1000  19547              0.211
47  1000  19548              0.152
48  1000  19601              0.063
```

You can see that we have 48 measurements before changing to the second day of data (196) for `ID=1000`. And then, when `ID=1000` ends, we get to `ID=1001` and back again to `time=19501`.

Here you can see the last rows of the dataset:

```
            ID   Time  Consumption (kWh)
24465789  1999  72948              0.499
24465790  1999  73001              0.327
24465791  1999  73002              0.314
24465792  1999  73003              0.397
24465793  1999  73004              0.156
24465794  1999  73005              0.198
24465795  1999  73006              0.131
24465796  1999  73007              0.324
24465797  1999  73008              0.197
24465798  1999  73009              0.303
24465799  1999  73010              0.188
24465800  1999  73011              0.160
24465801  1999  73012              0.294
24465802  1999  73013              0.160
24465803  1999  73014              0.305
24465804  1999  73015              0.132
24465805  1999  73016              0.145
24465806  1999  73017              0.363
24465807  1999  73018              1.124
24465808  1999  73019              1.183
24465809  1999  73020              0.976
24465810  1999  73021              2.343
24465811  1999  73022              1.812
24465812  1999  73023              0.881
24465813  1999  73024              0.847
24465814  1999  73025              1.260
24465815  1999  73026              2.558
24465816  1999  73027              2.956
24465817  1999  73028              1.773
24465818  1999  73029              0.864
24465819  1999  73030              1.185
24465820  1999  73031              1.237
24465821  1999  73032              1.590
24465822  1999  73033              2.363
24465823  1999  73034              2.080
24465824  1999  73035              2.085
24465825  1999  73036              3.800
24465826  1999  73037              1.833
24465827  1999  73038              2.001
24465828  1999  73039              1.598
24465829  1999  73040              4.778
24465830  1999  73041              5.518
24465831  1999  73042              3.441
24465832  1999  73043              1.432
24465833  1999  73044              2.831
24465834  1999  73045              4.180
24465835  1999  73046              2.372
24465836  1999  73047              3.488
24465837  1999  73048              3.972
```

So, instead of having the `time` for each `id` (like in the example of Multiclass), i have `time` for several `days` for each `id`. We can see that the last day is `730` and since the first is `195`, we have `730-195 = 535` days of data for each `ID`. One thing to note is that some `ID` values are missing, not sure if this is a problem for tsfresh (in practice, for my problem, it does not matter if some `ID` are missing)

How can i use tsfresh in this case?

**Comment by nils-braun** [comment, marked as answer]
Hi @muriloasouza,
yes, it is possible to use tsfresh for your use-case! How exactly depends on how you want to use the extracted features later.
The `id` should correspond to the target you want to predict/classify later. This means, 
* if you want to predict one class for a single ID (for all days together), you can just use the data as it is and follow the tutorial basically word for word.
* if you want to predict a class per ID per day, you should generate a unique ID for each ID and day - something like concatenating the current ID with the day number for each row and using the concatenated ID for `tsfresh`. 
Please note that for `tsfresh`, the time column does not need to be in any time or date format. It is totally fine having an integer as in your case - the only constraint is that (a) it is sortable (which it is in your case) and (b) the time steps are rather equidistant (which they are also in your case).

**Reply by muriloasouza** [reply]
Great, thanks @nils-braun ! I am going to predict a class per ID per day.
